domains/transfer/Transfer.API/API/Transfer/Api/[Features]/GetTransferAgreementsQueryHandler.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Transfer.Api.Dto.Responses;
+using API.UnitOfWork;
+using DataContext.Models;
+using EnergyOrigin.Domain.ValueObjects;
+using MediatR;
+
+namespace API.Transfer.Api._Features_;
+
+public class GetTransferAgreementsQueryHandler(IUnitOfWork unitOfWork) : IRequestHandler<GetTransferAgreementsQuery, GetTransferAgreementQueryResult>
+{
+
+    public async Task<GetTransferAgreementQueryResult> Handle(GetTransferAgreementsQuery request, CancellationToken cancellationToken)
+    {
+        var transferAgreements =
+            await unitOfWork.TransferAgreementRepo.GetTransferAgreementsList(request.OrganizationId, request.ReceiverTin,  cancellationToken);
+        var transferAgreementProposals = await unitOfWork.TransferAgreementRepo.GetTransferAgreementProposals(request.OrganizationId, cancellationToken);
+
+        if (!transferAgreementProposals.Any() && !transferAgreements.Any())
+        {
+            return new GetTransferAgreementQueryResult(new());
+        }
+
+        var transferAgreementDtos = transferAgreements
+            .Select(x => new GetTransferAgreementQueryResultItem(x.Id, x.StartDate.EpochSeconds, x.EndDate?.EpochSeconds, x.SenderName.Value,
+                x.SenderTin.Value, x.ReceiverTin.Value, TransferAgreementTypeMapper.MapCreateTransferAgreementType(x.Type),
+                GetTransferAgreementStatusFromAgreement(x)))
+            .ToList();
+
+        var transferAgreementProposalDtos = transferAgreementProposals
+            .Select(x => new GetTransferAgreementQueryResultItem(x.Id, x.StartDate.EpochSeconds, x.EndDate?.EpochSeconds, string.Empty,
+                string.Empty, x.ReceiverCompanyTin?.Value, TransferAgreementTypeMapper.MapCreateTransferAgreementType(x.Type),
+                GetTransferAgreementStatusFromProposal(x)))
+            .ToList();
+
+        transferAgreementProposalDtos.AddRange(transferAgreementDtos);
+
+        return new GetTransferAgreementQueryResult(transferAgreementProposalDtos);
+    }
+
+    private TransferAgreementStatus GetTransferAgreementStatusFromProposal(TransferAgreementProposal transferAgreementProposal)
```

Comment:
This logic should be moved out into service / helper / whatever. Because if we forget to update it one place it will be miss alligned.